dataRetriever/dataPool/proofsCache/proofsCache.go

```diff
@@ -47,18 +47,34 @@ func (pc *proofsCache) addProof(proof data.HeaderProofHandler) {
 	pc.mutProofsCache.Lock()
 	defer pc.mutProofsCache.Unlock()
 
-	pc.proofsByNonce = append(pc.proofsByNonce, &proofNonceMapping{
-		headerHash: string(proof.GetHeaderHash()),
-		nonce:      proof.GetHeaderNonce(),
-	})
-
-	sort.Slice(pc.proofsByNonce, func(i, j int) bool {
-		return pc.proofsByNonce[i].nonce < pc.proofsByNonce[j].nonce
-	})
+	pc.insertProofByNonce(proof)
 
 	pc.proofsByHash[string(proof.GetHeaderHash())] = proof
 }
 
+func (pc *proofsCache) insertProofByNonce(proof data.HeaderProofHandler) {
+	if len(pc.proofsByNonceBuckets) == 0 {
+		pc.insertInNewBucket(proof)
+		return
+	}
+
+	headBucket := pc.proofsByNonceBuckets[0]
+
+	if headBucket.isFull() {
```

Comment:
in case of syncing node this will cause it to keep accumulating new nonces in each old nonces buckets, which will delay their cleanup.

Reply:
new model with range buckets should cover this case